pkg/cmd/cli/cmd/deploy_test.go

```diff
@@ -328,3 +329,66 @@ func TestDeploy_reenableTriggers(t *testing.T) {
 		}
 	}
 }
+
+// TestDeploy_cancelOnDifferentTZ tests that oc deploy --cancel sets the cancellation
+// timestamp based on the timezone of the server.
+func TestDeploy_cancelOnDifferentTZ(t *testing.T) {
+	now := time.Now()
+
+	// location of the client is New York
+	clientLoc, _ := time.LoadLocation("America/New_York")
+	clientTime := now.In(clientLoc)
+
+	// location of the server is Bangkok
+	serverLoc, _ := time.LoadLocation("Asia/Bangkok")
+	serverTime := now.In(serverLoc)
+
+	if serverTime.Format(time.RFC3339) == clientTime.Format(time.RFC3339) {
+		t.Fatalf("the test is not really testing anything")
+	}
+
+	nowFn = func() time.Time {
```

Comment:
As @smarterclayton [hinted](https://github.com/openshift/origin/pull/10802#issuecomment-249660881), this may accidentally affect tests running after this test, for the nowFun global is changed forever, no clean up. Also complicates running tests in parallel.